web/src/test/java/org/springframework/security/web/firewall/StrictHttpFirewallTests.java

```diff
@@ -595,4 +597,38 @@ public void getFirewalledRequestWhenUntrustedDomainThenException() {
 
 		this.firewall.getFirewalledRequest(this.request);
 	}
+
+	@Test(expected = RequestRejectedException.class)
+	public void getFirewalledRequestWhenControlCharacterInHeaderNameThenException() {
+		this.request.addHeader("Bad\0Name", "some value");
+
+		HttpServletRequest request = this.firewall.getFirewalledRequest(this.request);
+		request.getHeaderNames();
+	}
+
+	@Test(expected = RequestRejectedException.class)
+	public void getFirewalledRequestWhenControlCharacterInHeaderValueThenException() {
+		this.request.addHeader("Something", "bad\0value");
+
+		HttpServletRequest request = this.firewall.getFirewalledRequest(this.request);
+		request.getHeader("Something");
+	}
+
+	@Test(expected = RequestRejectedException.class)
+	public void getFirewalledRequestWhenControlCharacterInParameterNameThenException() {
+		this.request.addParameter("Bad\0Name", "some value");
+
+		HttpServletRequest request = this.firewall.getFirewalledRequest(this.request);
+		request.getParameterNames();
+	}
+
+	@Test(expected = RequestRejectedException.class)
+	public void getFirewalledRequestWhenNotAllowedInParameterValueThenException() {
+		this.firewall.setAllowedHostnames(value -> value.equals("bad value"));
```

Comment:
It doesn't make sense to me that you'd call `setAllowedHostnames` here since you are testing parameter values.

Reply:
Agreed, fixed.